### DISCUSSION TITLE
Zerorpc-python

### DISCUSSION THREAD

**Discussion by gsakkis** [Ananto30/zero#48] [General]
How is this project different from [zerorpc-python](https://github.com/0rpc/zerorpc-python/blob/master/zerorpc/patterns.py)?

**Comment by Ananto30** [comment]
Zero is type hinted and have code generation for clients, it supports async and use multicores. Also I have plan to use other options than zeromq (i.e. grpc) probably in future. It is more like a microservice oriented framework.

And zerorpc-python is an amazing library to call any modules and write your own. Probably the underlying implementation can be same but the intentions are different.

Question to you @gsakkis, what is the difference between flask and bottle? :) Your question sounded the same.

**Reply by gsakkis** [reply]
@Ananto30 thanks, the asyncio support and multicore utilization are indeed important differentiators. 

I think the question is quite reasonable given the almost identical project names, the common dependency on ZeroMQ and the one-line summaries:

`Zero is a simple Python framework (RPC like) to build fast and high performance microservices or distributed servers.`
vs
`Zerorpc is a light-weight, reliable and language-agnostic library for distributed communication between server-side processes.`

> Question to you @gsakkis, what is the difference between flask and bottle? :) 

Is this a rhetorical question or is it actually implied that `zero : zerorpc :: bottle : flask` (or vice versa)? Because many/most people would know how to choose between flask and bottle :)
